components/create-post-page/UploadTextbookCover.tsx

```diff
@@ -1,29 +1,61 @@
-import ImageUploadModal from "@components/ImageUpload/ImageUploadModal";
+import { Box, Button, Icon, VStack } from "@chakra-ui/react";
+import ImageUpload from "@components/ImageUpload/ImageUpload";
+import { RefreshIcon } from "@heroicons/react/outline";
+import { useState } from "react";
+import { v4 as uuid } from "uuid";
 
 export const TEXTBOOK_ASPECT_RATIO = 4 / 5;
 
 interface Props {
   onCoverPhotoUploaded: (coverPhoto: Blob) => void;
-  isOpen: boolean;
   onClose: VoidFunction;
 }
 
 export default function UploadTextbookCover({
   onCoverPhotoUploaded,
-  isOpen,
   onClose,
 }: Props) {
   const newWrapTextbookCoverUploaded = async (blob: Blob) => {
     onCoverPhotoUploaded(blob);
   };
+
+  const [imageUploadKey, setImageUploadKey] = useState("");
+
   return (
-    <ImageUploadModal
-      key={"image-upload-modal"}
-      aspectRatio={4 / 5}
-      isOpen={isOpen}
-      onClose={onClose}
+    <ImageUpload
+      key={imageUploadKey}
+      aspectRatio={TEXTBOOK_ASPECT_RATIO}
+      onCancel={onClose}
       onUpload={newWrapTextbookCoverUploaded}
       shape="rect"
-    />
+    >
+      {({ body, ConfirmationButtons }) => (
+        <VStack maxWidth="sm" margin="auto">
+          <Box width="full" minHeight="60">
+            {body}
+          </Box>
+          <Box width="full" textAlign="right">
+            <ConfirmationButtons
+              spacing="5"
+              CancelButton={(props) => (
+                <Button
+                  variant="link"
+                  leftIcon={<Icon as={RefreshIcon} />}
+                  {...props}
+                  onClick={() => setImageUploadKey(uuid())}
+                >
+                  Retake
```

Comment:
Consider only showing the Retake button if a user uploaded a photo.

![Screen Shot 2022-03-29 at 10 23 36 PM](https://user-images.githubusercontent.com/40670885/160739562-ea1e468d-273f-4e9d-93fd-08e74739e300.png)
.